tutorials/calibration_flow/gptq_like.py

```diff
@@ -0,0 +1,273 @@
+    return isinstance(mod, TwoLinear)
+
+# Adapted from https://github.com/pytorch/ao/pull/581
+def prepare_model_for_optimization_(model):
+    def forward_hook(
+        module,
+        args: Tuple[MultiTensor],
+        kwargs: Dict[str, MultiTensor],
+        output: Tuple[MultiTensor],
+    ):
+        # remove the hook to avoid recursive calls
+        module._forward_hook_handle.remove()
+        # in this case args will be a tuple of 2 MultiTensors since
+        # the TwoLinear module takes 2 inputs, and each input will collect a list of normal Tensors
+
+        # we can use these two MultiTensors to calculate the quantization parameters for each input Tensor
+        act_obs1 = AffineQuantizedMinMaxObserver(MappingType.ASYMMETRIC, torch.uint8, granularity_type=PerTensor(), eps=torch.finfo(torch.float32).eps, scale_dtype=torch.float32, zero_point_dtype=torch.int32)
+        for inp in args[0]:
+            act_obs1(inp)
+
+        act_obs2 = AffineQuantizedMinMaxObserver(MappingType.ASYMMETRIC, torch.uint8, granularity_type=PerTensor(), eps=torch.finfo(torch.float32).eps, scale_dtype=torch.float32, zero_point_dtype=torch.int32)
+        for inp in args[1]:
+            act_obs2(inp)
+
+        input_scale1, input_zp1 = act_obs1.calculate_qparams()
+        input_scale2, input_zp2 = act_obs2.calculate_qparams()
+
+        # we can optimize/modify the module here
+        module.input_scale1 = input_scale1
+        module.input_zp1 = input_zp1
+        module.input_scale2 = input_scale2
+        module.input_zp2 = input_zp2
+
+        # rerun the module with quantized and dequantized inputs
+        new_input1 = []
+        for inp in args[0]:
+            new_input1.append(fake_quantize_affine(inp, inp.shape, input_scale1, input_zp1, torch.uint8))
+
+        new_input2 = []
+        for inp in args[1]:
+            new_input2.append(fake_quantize_affine(inp, inp.shape, input_scale2, input_zp2, torch.uint8))
+
+        mt1 = MultiTensor(new_input1)
+        mt2 = MultiTensor(new_input2)
+
+        output_with_fake_quantized_inputs = module(mt1, mt2)
+        # we can return the modified output so it can be consumed by future modules
+        return output_with_fake_quantized_inputs
+
+    def _register_forward_hook(module: torch.nn.Module):
+        forward_hook_handle = module.register_forward_hook(
+            forward_hook, with_kwargs=True
+        )
+        module._forward_hook_handle = forward_hook_handle
+        return module
+
+    _replace_with_custom_fn_if_matches_filter(
+        model, _register_forward_hook, _is_two_linear
+    )
+
+# using a function to align with the API in quant_api
+def apply_activation_static_quant():
+    def _apply_activation_static_quant(observed_two_linear):
+        target_dtype = torch.uint8
+
+        linear1 = observed_two_linear.linear1
+        # activation quantization
+        act_scale, act_zero_point = observed_two_linear.input_scale1, observed_two_linear.input_zp1
```

Comment:
I'm not sure I follow why this assignment is needed 
and I'm not sure i understand why `input_quant_func1` is needed? and why those couldn't be arguments to `to_linear_activation_quantized` I haven't carefully reviewed AQT so would appreciate some more context on what's the intent here

Reply:
`input_quant_func1` is a general input quantization function that can be used to quantize activations, we can't hard code the args for `to_linear_activation_quantized` since the `input_quant_func` can be any function, it's not restricted to be using affine quantization here

we don't have docs for `to_linear_activation_quantized`, this is used to implement static and dynamic quantization for activations. I can add some docs for this